### PR TITLE
Visual ANALYZE

### PR DESCRIPTION
**Pull request by hendrikmakait**
Utility to analyze distribution of partition sizes (and other cost in the future).

Example output:
analyze-operation-279b50339626b773def601152c7db754

### CODE REVIEW

**Comment by mrocklin** [comment]
Thoughts on maybe also getting runtime durations while we're at it?

File: dask_expr/diagnostics/_analyze_plugin.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    _scheduler: Scheduler | None
+
+    def __init__(self) -> None:
+        self._scheduler = None
+
+    async def start(self, scheduler: Scheduler) -> None:
+        self._scheduler = scheduler
+        scheduler.handlers["analyze_get_statistics"] = self.get_statistics
+        worker_plugin = _AnalyzeWorkerPlugin()
+        await self._scheduler.register_worker_plugin(
```
[the remark this answers is not in the document]
Comment (hendrikmakait) [reply dask_expr/diagnostics/_analyze_plugin.py:21]: I think we have most of the information on the task groups, but I lack a reliable way for mapping task groups to expressions. All of this should look much better once we have expressions on the scheduler. As a nice side effect, I can inline these calculations into fused expressions.

> Thoughts on maybe also getting runtime durations while we're at it?

This is currently blocked on the lack of a reliable mapping (as mentioned above).

**Comment by hendrikmakait** [comment]
For context, as opposed to #885, this is much more experimental and first and foremost a utility for me to investigate performance on TPC-H benchmarks.

File: dask_expr/diagnostics/_analyze_plugin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrocklin) [reply dask_expr/diagnostics/_analyze_plugin.py:21]: I would be surprised if each expression didn't result in exactly one task group with the same name.

**Comment by hendrikmakait** [comment]
This still lacks a test. Otherwise, it's good to go. At a later point in time, the code should be cleaned up and integrated better, but that's nothing I want to worry about right now.

**Pull request merged**